### ISSUE TITLE
Download single file and download from project directory

### ISSUE THREAD

**Issue by ArvidBaa** [humy2833/FTP-Simple#61]
I often work with files on the ftp server that also gets edited by other people. Because of that I usually need to download a single file, or the entire project directory. I also have multiple project directories configured at the same time, which works perfectly with the recent update with save without confirm for project directories. I want to thank you for that update, by the way. It was a great improvement for me.

To my issue. When I want to download a single file, or the entire project directory from the server, I find it hard to do so. I have to search through the server, starting in the base path defined in the config-file, and eventually find the folder or file I need. This is a little frustrating. Is there some setting or command that I have missed that could make this easier? I would like to be prompted with the project directory immediately (just as when choosing the "Diff" option), when I want to download all files and folders, and only the active and opened file when I want to download a single file. It would also be awesome if these options were available in the right-click-dropdown alongside with the "Delete", "Diff" and "Save" options.

**Comment by rajeshshou** [comment]
Same here

**Comment by humy2833** [comment]
You can specify the server base path with the "path" option in the configuration file.
You can also use the "project" option to match the local project path to the server's project path.

**Comment by ArvidBaa** [comment]
Yes, I know about the path and working directory settings. I have done that and it's doing wonders when I want to save and upload, but it does nothing when I want to download.

**Comment by humy2833** [comment]
I can not speak English very well. So I am getting help from Google Translator.
So I was late to understand your words. Sorry.
Is your request "the ability to locally overwrite files on the server without manual traversal"?

**Comment by ArvidBaa** [comment]
Yeah, I think that would be a good description of my issue.

I want two things:
1) Download the current project directory without manual traversal.
2) Download the current open file without manual traversal.

Thanks in advance! You are doing a great job with this extension!

**Comment by ArvidBaa** [comment]
I have noticed that the latest update included a change, so that the "download" option now automatically downloads the open and active file, and that the option has been added to the right-click-menu for each file and folder. That is really awesome, in my opinion!

On simple addition to this, though, that I would like, is to add a new option in the dropdowns, to download the entire project folder. Right now I have to find some empty space, or the scrollbar, in the explorer field to be able to download the entire project folder. (Perhaps it is possible to get to that option in some other way, but I haven't had time to look deeper right now. Maybe someone has some tips?)

On top of this, I have also added a feature request for sync, besides the download option, so that only files out of sync are downloaded. But I still want the download option too!

**Comment by ArvidBaa** [comment]
I just realized that it doesn't work to download the project directory as I mentioned earlier. The effect is just that the entire project directory gets downloaded as a subfolder in the project directory. That is not the expected behavior.

**Comment by ArvidBaa** [comment]
Could you please have another look at this issue. There is no way to download the entire root folder anymore, just subfolders and individual files. Simply put, I still want my request 1. Download the current project directory without manual traversal.

**Comment by ArvidBaa** [comment]
Could you please consider reopening this ticket?
